controller/multi_writer_at.go

```diff
@@ -80,9 +82,27 @@ func (m *MultiWriterAt) WriteAt(p []byte, off int64) (n int, err error) {
 		errors.Updaters = m.updaters
 		errors.QuorumErrors = quorumErrs
 	}
-
+	//Below code is introduced to make sure that the IO has been written to more
+	//than 50% of the replica.
+	//If any replica has errored, return with the length of data written and the
+	//erroed replica details so that it can be closed.
 	if replicaErrored || quorumErrored {
-		return 0, &errors
+		for _, err1 := range replicaErrs {
+			if err1 != nil {
+				replicaErrCount++
+			}
+		}
+		for _, err1 := range quorumErrs {
+			if err1 != nil {
+				quorumErrCount++
+			}
+		}
+		if (len(m.writers)-replicaErrCount >= len(m.writers)/2) &&
```

Comment:
should it be > instead of >= as in case of 4 replica we want IO to be successful on n/2 + 1 replica i.e. 3?

Reply:
Resolved in latest commit and ci test case is also being rechecked to figure out why this issue was not caught.

Reply:
There was no test case for this condition to check for failure.